alibi_detect/tests/test_dep_management.py

```diff
@@ -66,8 +66,8 @@ def test_cd_torch_dependencies(opt_dep):
 
     dependency_map = defaultdict(lambda: ['default'])
     for dependency, relations in [
-        ("HiddenOutput", ['torch']),
-        ("preprocess_drift", ['torch'])
+        ("HiddenOutput", ['torch', 'keops']),
```

Comment:
I don't really understand why `keops` is added for all these tests that are supposed to test `torch` dependencies?

Reply:
The `keops` bucket includes `torch` as the `keops` functionality needs it. Therefore any functionality dependent on `torch` will also pass if the user has the `keops` bucket installed.

Reply:
So are these tests now run twice, once with `[torch]` and once with `[keops]` installed?

Reply:
The `test_dep_management` tests are run once for every optional dependency bucket. All they do is check if the correct object is imported dependent on the optional dependencies present. So this one says that we should receive a `MissingOptionalDependecy` object for everything except `all`, `torch` and `keops`. As they're redundant in the full test suite we skip them there.

Reply:
It's just these tests though and they're reasonably fast. The only thing that takes time is building the tox environments.

Reply:
@jklaise happy with this?